catalystwan/models/policy/definition/url_filtering.py

```diff
@@ -0,0 +1,152 @@
+    "dating",
+    "government",
+    "gross",
+    "hacking",
+    "hate-and-racism",
+    "health-and-medicine",
+    "home",
+    "hunting-and-fishing",
+    "illegal",
+    "image-and-video-search",
+    "individual-stock-advice-and-tools",
+    "internet-communications",
+    "internet-portals",
+    "job-search",
+    "keyloggers-and-monitoring",
+    "kids",
+    "legal",
+    "local-information",
+    "malware-sites",
+    "marijuana",
+    "p2p",
+    "parked-sites",
+    "pay-to-surf",
+    "personal-sites-and-blogs",
+    "philosophy-and-political-advocacy",
+    "phishing-and-other-frauds",
+    "private-ip-addresses",
+    "proxy-avoid-and-anonymizers",
+    "questionable",
+    "real-estate",
+    "recreation-and-hobbies",
+    "reference-and-research",
+    "religion",
+    "search-engines",
+    "sex-education",
+    "shareware-and-freeware",
+    "shopping",
+    "social-network",
+    "society",
+    "sports",
+    "spam-urls",
+    "spyware-and-adware",
+    "streaming-media",
+    "swimsuits-and-intimate-apparel",
+    "training-and-tools",
+    "translation",
+    "travel",
+    "uncategorized",
+    "unconfirmed-spam-sources",
+    "violence",
+    "weapons",
+    "web-advertisements",
+    "web-based-email",
+    "web-hosting",
+    "open-http-proxies",
+    "online-personal-storage",
+    "online-greeting-cards",
+    "nudity",
+    "news-and-media",
+    "music",
+    "motor-vehicles",
+    "military",
+]
+
+
+class UrlFilteringDefinition(BaseModel):
+    model_config = ConfigDict(populate_by_name=True)
+    web_categories_action: WebCategoriesAction = Field(
+        validation_alias="webCategoriesAction", serialization_alias="webCategoriesAction"
+    )
+    web_categories: List[WebCategories] = Field(validation_alias="webCategories", serialization_alias="webCategories")
+    web_reputation: WebReputation = Field(validation_alias="webReputation", serialization_alias="webReputation")
+    url_white_list: Optional[Reference] = Field(
+        default=None, validation_alias="urlWhiteList", serialization_alias="urlWhiteList"
+    )
+    url_black_list: Optional[Reference] = Field(
+        default=None, validation_alias="urlBlackList", serialization_alias="urlBlackList"
+    )
+    block_page_action: BlockPageAction = Field(
+        validation_alias="blockPageAction", serialization_alias="blockPageAction"
+    )
+    block_page_contents: str = Field(
+        default=BLOCK_PAGE_CONTENT_HEADER, validation_alias="blockPageContents", serialization_alias="blockPageContents"
+    )
+    logging: List[str] = Field(default=[])
+    enable_alerts: bool = Field(validation_alias="enableAlerts", serialization_alias="enableAlerts")
+    alerts: Set[UrlFilteringAlerts] = Field(default=[], validation_alias="alerts", serialization_alias="alerts")
+    target_vpns: List[VpnId] = Field(default=[], validation_alias="targetVpns", serialization_alias="targetVpns")
+
+    @field_validator("url_black_list", "url_white_list", mode="before")
```

Comment:
I am not sure, if it's the best solution.
in some cases the url_black_list / white_list is  not None, but empty dict, which cannot be deserialised do the Reference object.

The validator checks if value is an empty dict and the returns None,.

Everything works as expected. Tested a few cases